File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
# 최대공약수 구하는 함수
def gcd(x, y):
    while(True):
        if x % y == 0:
            return y
        else:
            t = x % y
            x = y
            y = t
            
# 최소공배수 구하는 함수
def lcm(x, y):
    return x * y // gcd(x, y)

n, m = map(int, input().split())

if n > m:
    print(gcd(n, m))
    print(lcm(n, m))
else:
    print(gcd(m, n))
    print(lcm(m, n))

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        
        answer[0] = gcd(n, m);
        answer[1] = lcm(n, m);
        
        return answer;
    }
    
    int gcd(int a, int b) { // 최대공약수
        
        int tmp;
        
        if(a < b) {
            tmp = a;
            a = b;
            b = tmp;
        }
        
        while(b != 0) {
            tmp = a % b;
            a = b;
            b = tmp;
        }
        
        return a;
    }
    
    int lcm(int a, int b) { // 최소 공배수
        
        return (a * b) / gcd(a, b);
    }
}